.circleci/config.yml

```diff
@@ -0,0 +1,132 @@
+version: 2
+
+variables:
+  restore_cache: &restore_cache
+    restore_cache:
+      keys:
+        - miniconda-{{ .Environment.MINICONDA_VER }}-{{ .Environment.BIOCONDA_UTILS_TAG }}-{{ arch }}
+  save_cache: &save_cache
+    save_cache:
+      key: miniconda-{{ .Environment.MINICONDA_VER }}-{{ .Environment.BIOCONDA_UTILS_TAG }}-{{ arch }}
+      paths:
+        - /tmp/workspace/miniconda
+  setup: &setup
+    run: 
+      name: Setup bioconda-utils
+      command: .circleci/setup.sh
+  macos: &macos
+    macos:
+      xcode: "9.0"
+  linux: &linux
+    machine: true
+
+
+jobs:
+  lint:
+    <<: *linux
+    steps:
+      - *restore_cache
+      - checkout
+      - *setup
+      - *save_cache
+      - run:
+          name: Linting
+          command: bioconda-utils lint recipes config.yml --git-range master HEAD
+
+  # build and test on linux
+  test-linux:
+    <<: *linux
+    steps:
+      - *restore_cache
+      - checkout
+      - *setup
+      - *save_cache
+      - run:
+          name: Building and testing
+          command: bioconda-utils build recipes config.yml --git-range master HEAD --docker --mulled-test
+
+  # build and test on macos
+  test-macos:
+    <<: *macos
+    steps:
+      - checkout
+      # We cannot cache the miniconda installation on macos.
+      # The reason is that tar seems to have a bug when handling sym and hardlinks in the miniconda installation.
+      # The error is, e.g.: Cannot extract through symlink tmp/workspace/miniconda/LICENSE.rst
```

Comment:
If this is due to hard-/symlinks created by the `conda install` command, we could try `conda install --copy` during the setup. If this happen because of files/links Miniconda itself already installed, then nevermind.

Reply:
I think it is both, see [here](https://circleci.com/gh/bioconda/bioconda-recipes/73?utm_campaign=vcs-integration-link&utm_medium=referral&utm_source=github-build-link)

We could still try to manually copy those symlinks away before the cache is created.

Reply:
Ah, thanks for pointing me at the log. I think it's something non-Conda related as it even shows those errors for the cached `repodata.json` files which are very unlikely to be linked somewhere else:
```
tmp/workspace/miniconda/pkgs/cache/: Cannot extract through symlink tmp/workspace/miniconda/pkgs/cache
tmp/workspace/miniconda/pkgs/cache/08159f17.json: Cannot extract through symlink tmp/workspace/miniconda/pkgs/cache/08159f17.json
tmp/workspace/miniconda/pkgs/cache/09cdf8bf.json: Cannot extract through symlink tmp/workspace/miniconda/pkgs/cache/09cdf8bf.json
tmp/workspace/miniconda/pkgs/cache/09cdf8bf.q: Cannot extract through symlink tmp/workspace/miniconda/pkgs/cache/09cdf8bf.q
tmp/workspace/miniconda/pkgs/cache/0cdb2265.json: Cannot extract through symlink tmp/workspace/miniconda/pkgs/cache/0cdb2265.json
tmp/workspace/miniconda/pkgs/cache/0cdb2265.q: Cannot extract through symlink tmp/workspace/miniconda/pkgs/cache/0cdb2265.q
tmp/workspace/miniconda/pkgs/cache/2a957770.json: Cannot extract through symlink tmp/workspace/miniconda/pkgs/cache/2a957770.json
tmp/workspace/miniconda/pkgs/cache/2a957770.q: Cannot extract through symlink tmp/workspace/miniconda/pkgs/cache/2a957770.q
tmp/workspace/miniconda/pkgs/cache/31ce02e0.json: Cannot extract through symlink tmp/workspace/miniconda/pkgs/cache/31ce02e0.json
tmp/workspace/miniconda/pkgs/cache/31ce02e0.q: Cannot extract through symlink tmp/workspace/miniconda/pkgs/cache/31ce02e0.q
tmp/workspace/miniconda/pkgs/cache/35adf087.json: Cannot extract through symlink tmp/workspace/miniconda/pkgs/cache/35adf087.json
tmp/workspace/miniconda/pkgs/cache/35adf087.q: Cannot extract through symlink tmp/workspace/miniconda/pkgs/cache/35adf087.q
tmp/workspace/miniconda/pkgs/cache/716b88c0.json: Cannot extract through symlink tmp/workspace/miniconda/pkgs/cache/716b88c0.json
tmp/workspace/miniconda/pkgs/cache/87d00fe333551440c29e16192807b6e6.png: Cannot extract through symlink tmp/workspace/miniconda/pkgs/cache/87d00fe333551440c29e16192807b6e6.png
tmp/workspace/miniconda/pkgs/cache/9779607c273dc0786bd972b4cb308b58.png: Cannot extract through symlink tmp/workspace/miniconda/pkgs/cache/9779607c273dc0786bd972b4cb308b58.png
tmp/workspace/miniconda/pkgs/cache/9ca791dd.json: Cannot extract through symlink tmp/workspace/miniconda/pkgs/cache/9ca791dd.json
tmp/workspace/miniconda/pkgs/cache/9ca791dd.q: Cannot extract through symlink tmp/workspace/miniconda/pkgs/cache/9ca791dd.q
tmp/workspace/miniconda/pkgs/cache/abae4451.json: Cannot extract through symlink tmp/workspace/miniconda/pkgs/cache/abae4451.json
tmp/workspace/miniconda/pkgs/cache/abae4451.q: Cannot extract through symlink tmp/workspace/miniconda/pkgs/cache/abae4451.q
tmp/workspace/miniconda/pkgs/cache/d85a531e.json: Cannot extract through symlink tmp/workspace/miniconda/pkgs/cache/d85a531e.json
tmp/workspace/miniconda/pkgs/cache/d85a531e.q: Cannot extract through symlink tmp/workspace/miniconda/pkgs/cache/d85a531e.q
tmp/workspace/miniconda/pkgs/cache/df7feebede9861a203b480c119e38b49.png: Cannot extract through symlink tmp/workspace/miniconda/pkgs/cache/df7feebede9861a203b480c119e38b49.png
```
So maybe a `/tmp` is symlinked in CircleCI's macOS setup?